File: 2º/PDOO/Practias/Napakalaki-ruby/lib/monster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "prize.rb"
require_relative "bad_consequence.rb"
require_relative "treasure_kind.rb"

class Monster
  attr_accessor :name
  attr_accessor :combatLevel
  attr_accessor :prize
  attr_accessor :bc
  
  def initialize(name, combatLevel, prize, bc)
    @name = name
    @combatLevel = combatLevel
    @prize = prize
    @bc = bc
    
  end
  def to_s
    "Nombre: #{@name} Nivel: #{@combatLevel} Premio: #{@prize} Mal rollo: #{@bc}"
  end
  
  def getLevelsGained
    @prize.level
  end

  def getTreasuresGained
    @prize.treasures
  end
  
end

File: 2º/PDOO/Practias/Napakalaki/src/napakalaki/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

/**
 *
 * @author arturo
 */
public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence bc;        
    
    Monster(String name, int level, BadConsequence bc, Prize prize){
        this.name=name;
        this.combatLevel=level;
        this.bc=bc;
        this.prize=prize;        
    }

    public String getName() {
        return name;
    }

    public int getCombatLevel() {
        return combatLevel;
    }

    public Prize getPrize() {
        return prize;
    }

    public BadConsequence getBc() {
        return bc;
    }
    
    public int getLevelsGained(){
        return prize.getLevel();
    }
    
    public int getTreasuresGained(){
        return prize.getTreasures();
}

    @Override
    public String toString() {
        return "Monster{" + "name=" + name + ", combatLevel=" + combatLevel + ", prize=" + prize + ", bc=" + bc + '}';
    }

    
    
}
